assets/course-theme/adminbar-layout.js

```diff
@@ -15,6 +20,15 @@ const trackAdminbarOffset = () => {
 		passive: true,
 	} );
 
+	/**
+	 * The debounce has 2 reasons here:
+	 * 1. Reduce the number of times we call the function in a resize.
+	 * 2. The admin bar contains an animated transition, so this transition
+	 *    needs to be completed in order to make the correct calc.
+	 */
+	// eslint-disable-next-line @wordpress/no-global-event-listener
+	window.addEventListener( 'resize', debounce( updateAdminbarOffset, 500 ) );
```

Comment:
I'm not sure we should care about resizing the window. We often test responsiveness that way, but the real-world use case is just people having a smaller screen. AFAIK `lodash` doesn't support tree-shaking so for this debounce we're bringing that entire library to our frontend, which is probably not worth it.

Reply:
I agree. I don't think it's worth supporting a resize change. I don't think anyone resizes their browser after visiting a website. Only developers who need to support responsive layout do that. 😆

Reply:
Good catch! It wasn't in our bundle, but it was being [included through WP lodash](https://www.npmjs.com/package/@wordpress/dependency-extraction-webpack-plugin). But yeah! We don't need to import it there. Updated it here: https://github.com/Automattic/sensei/pull/4684/commits/836dae5c4422b49c8d37e0dda023f1672470f298

![Screen Shot 2022-01-27 at 11 04 41](https://user-images.githubusercontent.com/876340/151374201-ed1a3e45-0cd0-46e0-9464-b8831a933f91.png)

I just don't agree that we shouldn't fix it for resize, since it's just one extra event not introducing a big complexity. I think if we can fix it easily, we should do it, and with that we avoid weird issues if the user decides to resize the window. Unless someone see any con on that, I think we should keep that.

> If we end up keeping a handler on resize, adding the { capture: false, passive: true } options to the addEventListener might improve performance, making it kind of like a non-blocking call. (Not sure if it's relevant for the resize event)

Since that documentation is specifically about the scrolling, I'll not add that for now. Probably it won't be a big deal since we're doing that with the debounce on the resize. But if we notice any performance issue, we can try that. ;)

Reply:
> with that we avoid weird issues if the user decides to resize the window.

On the other hand, if we make this page resilient to window resize, then user will expect all pages to behave this way. So they will go ahead and complain if other pages break when the window is resized. 🤷 

I think it's better to fix it if/when we have real complains about it. Because otherwise we are adding logic based on our own speculation instead of real world usage data.

Reply:
> On the other hand, if we make this page resilient to window resize, then user will expect all pages to behave this way. So they will go ahead and complain if other pages break when the window is resized. 🤷

I don't think it's a problem because other pages don't have any logic basic on that. It's something specific we have in the Learning Mode.

> I think it's better to fix it if/when we have real complains about it. Because otherwise we are adding logic based on our own speculation instead of real world usage data.

If it would introduce some big complexity to the code, I could agree with you, but considering it causes a real issue, and it's very easy to fix (just registering an extra event), I don't see a reason to not fix it. 🤔

Reply:
Yeah, I think you're right. It wouldn't hurt us to add this. 👍